### PR TITLE
style(index): add JSDOC

### PR DESCRIPTION
**Pull request by hasezoey**
Add JSDOC to all public functions

WIP because i am not sure for some descriptions

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -303,6 +343,16 @@ function _handleWrapError(instance, error, name, context, args, options, callbac
   }
 }
 
+/**
+ *
```
Comment (hasezoey) [inline index.js:347]: i dont quite know what this function does, so i left it empty

**Review by hasezoey — COMMENTED** [review]
some comments where i was not sure at, also:
- should the `context` parameters be of a specific type?

**Comment by hasezoey** [comment]
un-drafted this PR because only one thing is "TBD"(to be done), which is to give a jsdoc description of a function (see review-comments above)

File: index.js (hunk 1 above)
Comment (vkarpov15) [reply index.js:347]: I realize this function is a bit tricky to read and the name could be better.

This function executes pre hooks, then calls the wrapped function, then executes post hooks. So this function or `createWrapper()` (which just calls this function) is how we recommend executing hooks, unless you want more fine-grained control and want to call `execPre()` and `execPost()` yourself.

**Review by vkarpov15 — APPROVED** [review]

**Pull request merged**